cla-backend-go/users/repository.go

```diff
@@ -458,6 +460,110 @@ func (repo repository) GetUserByUserName(userName string, fullMatch bool) (*mode
+	if err != nil {
+		log.Warnf("error building expression for lf_email : %s, error: %v", userEmail, err)
+		return nil, err
+	}
+
+	// Assemble the query input parameters
+	queryInput := &dynamodb.QueryInput{
+		ExpressionAttributeNames:  expr.Names(),
+		ExpressionAttributeValues: expr.Values(),
+		KeyConditionExpression:    expr.KeyCondition(),
+		ProjectionExpression:      expr.Projection(),
+		TableName:                 aws.String(tableName),
+		IndexName:                 aws.String("lf-email-index"),
+	}
+
+	// Make the DynamoDB Query API call
+	result, err := repo.dynamoDBClient.Query(queryInput)
+	if err != nil {
+		log.Warnf("error retrieving user by lf_email: %s, error: %+v", userEmail, err)
+		return nil, err
+	}
+
+	// The user model
+	var dbUserModels []DBUser
+
+	err = dynamodbattribute.UnmarshalListOfMaps(result.Items, &dbUserModels)
+	if err != nil {
+		log.Warnf("error unmarshalling user record from database for lf_email: %s, error: %+v", userEmail, err)
+		return nil, err
+	}
+
+	if len(dbUserModels) == 0 {
+		return nil, nil
+	} else if len(dbUserModels) > 1 {
+		log.Warnf("retrieved %d results for the lf_email query when we should return 0 or 1", len(dbUserModels))
+	}
+
+	return convertDBUserModel(dbUserModels[0]), nil
+}
+
+// GetUserByGitHubUsername fetches the user record by github username
+func (repo repository) GetUserByGitHubUsername(gitHubUsername string) (*models.User, error) {
+	tableName := fmt.Sprintf("cla-%s-users", repo.stage)
+
+	// This is the key we want to match
+	condition := expression.Key("user_github_username").Equal(expression.Value(gitHubUsername))
+
+	// These are the columns we want returned
+	projection := buildUserProjection()
+
+	// Use the nice builder to create the expression
+	expr, err := expression.NewBuilder().WithKeyCondition(condition).WithProjection(projection).Build()
+	if err != nil {
+		log.Warnf("error building expression for user_github_username : %s, error: %v", gitHubUsername, err)
+		return nil, err
+	}
+
+	// Assemble the query input parameters
+	queryInput := &dynamodb.QueryInput{
+		ExpressionAttributeNames:  expr.Names(),
+		ExpressionAttributeValues: expr.Values(),
+		KeyConditionExpression:    expr.KeyCondition(),
+		ProjectionExpression:      expr.Projection(),
+		TableName:                 aws.String(tableName),
+		IndexName:                 aws.String("github-username-index"),
+	}
+
+	// Make the DynamoDB Query API call
+	result, err := repo.dynamoDBClient.Query(queryInput)
+	if err != nil {
+		log.Warnf("error retrieving user by user_github_username: %s, error: %+v", gitHubUsername, err)
+		return nil, err
+	}
+
+	// The user model
+	var dbUserModels []DBUser
+
+	err = dynamodbattribute.UnmarshalListOfMaps(result.Items, &dbUserModels)
+	if err != nil {
+		log.Warnf("error unmarshalling user record from database for user_github_username: %s, error: %+v", gitHubUsername, err)
+		return nil, err
+	}
+
+	if len(dbUserModels) == 0 {
```

Comment:
If we don't found user then we should return error 
ErrUserNotFound = "user not found"

It is common pattern. Many time we don't check if pointer is non nil or not,  if err is nil.